src/viewer/ImageViewer.jsx

```diff
@@ -235,32 +212,35 @@ export default class ImageViewer extends Component {
     if (this.state.canceled) {
       return <NoNetworkViewer onReload={this.reload} />
     }
-    const file = {
-      ...this.props.file,
-      links: this.props.file.links || this.state.links
-    }
+    const { file } = this.props
     const { scale, offsetX, offsetY } = this.state
     const style = {
       transform: `scale(${scale}) translate(${offsetX}px, ${offsetY}px)`
     }
     return (
       <div className={styles['pho-viewer-imageviewer']}>
-        {(this.state.needLinks || this.state.loading) && (
+        {this.state.loading && (
           <Spinner size="xxlarge" middle noMargin color="white" />
         )}
-        {!this.state.needLinks &&
-          file && (
-            <ImageLoader
-              photo={file}
-              src={`${cozy.client._url}${file.links.large}`}
-              style={style}
-              ref={photo => {
-                this.photo = React.findDOMNode(photo)
-              }}
-              onLoad={this.onImageLoad}
-              onError={this.onImageError}
-            />
-          )}
+        {file && (
+          <ImageLoader
+            file={file}
+            size="large"
+            ref={photo => {
+              this.photo = React.findDOMNode(photo)
+            }}
+            onError={this.onImageError}
+            key={file.id}
```

Comment:
This one is interesting : `ImageLoader` needs to reset it's internal state when a prop changes, so that the new image gets loaded. But 
- calling `setState` in `ComponentDidUpdate` is disallowed by our linter, even with a condition
- `componentWill*` lifecycle methods have been deprecated
- I ended up [reading this](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#alternative-2-reset-uncontrolled-component-with-an-instance-method) and using a `key`